source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -0,0 +1,605 @@
+    assert(Res == UR_RESULT_SUCCESS);
+
+    std::vector<char> KernelNameBuf(KernelNameSize);
+    Res = context.urDdiTable.Kernel.pfnGetInfo(
+        Kernel, UR_KERNEL_INFO_FUNCTION_NAME, KernelNameSize,
+        KernelNameBuf.data(), nullptr);
+    assert(Res == UR_RESULT_SUCCESS);
+
+    return std::string(KernelNameBuf.data(), KernelNameSize - 1);
+}
+
+ur_result_t SanitizerInterceptor::allocShadowMemory(
+    ur_context_handle_t Context, std::shared_ptr<DeviceInfo> &DeviceInfo) {
+    if (DeviceInfo->Type == DeviceType::CPU) {
+        context.logger.error("Unsupport device type");
+        return UR_RESULT_ERROR_INVALID_ARGUMENT;
+    } else if (DeviceInfo->Type == DeviceType::GPU_PVC) {
+        /// SHADOW MEMORY MAPPING (PVC, with CPU 47bit)
+        ///   Host/Shared USM : 0x0              ~ 0x0fff_ffff_ffff
+        ///   ?               : 0x1000_0000_0000 ~ 0x1fff_ffff_ffff
+        ///   Device USM      : 0x2000_0000_0000 ~ 0x3fff_ffff_ffff
+        constexpr size_t SHADOW_SIZE = 1ULL << 46;
+
+        // TODO: Protect Bad Zone
+        auto Result = context.urDdiTable.VirtualMem.pfnReserve(
+            Context, nullptr, SHADOW_SIZE, (void **)&DeviceInfo->ShadowOffset);
+        if (Result != UR_RESULT_SUCCESS) {
+            context.logger.error("Failed to allocate shadow memory on PVC: {}",
+                                 Result);
+            return Result;
+        }
+
+        DeviceInfo->ShadowOffsetEnd = DeviceInfo->ShadowOffset + SHADOW_SIZE;
+    } else {
+        context.logger.error("Unsupport device type");
+        return UR_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+    context.logger.info("ShadowMemory(Global): {} - {}",
+                        (void *)DeviceInfo->ShadowOffset,
+                        (void *)DeviceInfo->ShadowOffsetEnd);
+    return UR_RESULT_SUCCESS;
+}
+
+ur_result_t SanitizerInterceptor::enqueueMemSetShadow(
+    ur_context_handle_t Context, ur_device_handle_t Device,
+    ur_queue_handle_t Queue, uptr Ptr, uptr Size, u8 Value,
+    ur_event_handle_t DepEvent, ur_event_handle_t *OutEvent) {
+
+    uint32_t NumEventsInWaitList = DepEvent ? 1 : 0;
+    const ur_event_handle_t *EventsWaitList = DepEvent ? &DepEvent : nullptr;
+
+    auto ContextInfo = getContextInfo(Context);
+    auto DeviceInfo = ContextInfo->getDeviceInfo(Device);
+
+    if (DeviceInfo->Type == DeviceType::CPU) {
+        context.logger.error("Unsupport device type");
+        return UR_RESULT_ERROR_INVALID_ARGUMENT;
+    } else if (DeviceInfo->Type == DeviceType::GPU_PVC) {
+        ur_event_handle_t InternalEvent{};
+        ur_event_handle_t *Event = OutEvent ? OutEvent : &InternalEvent;
+
+        uptr ShadowBegin = MemToShadow_PVC(DeviceInfo->ShadowOffset, Ptr);
+        uptr ShadowEnd =
+            MemToShadow_PVC(DeviceInfo->ShadowOffset, Ptr + Size - 1);
+
+        {
+            static const size_t PageSize = [Context, Device]() {
+                size_t Size;
+                auto Result =
```

Comment:
`Result` is only used in non Release builds so is causing our CI to fail.

Since this is a immediately invoked lamba function, I think `Result` should be hoisted outside of here so that when an error occurs it can be returned from `Sanitizer::Interceptor::enqueueMemSetShadow`.

Either that or mark this as `[[maybe_unused]]` to silence the warning (treated as error) which is causing the CI to fail.

Reply:
Thanks, I've added `[[maybe_unused]]`